content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -115,13 +117,13 @@ Use the `@CordaInject` annotation to define a field to be set by Corda before th
 
 In this sample CorDapp, add these services:
 
-* `FlowEngine`
-* `FlowIdentity`
-* `FlowMessaging`
-* `TransactionBuilderFactory`
-* `IdentityService`
-* `NotaryLookupService`
-* `JsonMarshallingService`
+* `FlowEngine`: Provides basic control over a flow it is injected into, such as calling subflows, performing asynchronous tasks and putting the flow to sleep for a period of time.
```

Comment:
none of these discuss *why* I'd want to import them. The description is good, but itr's missing the obvious next step of "because we need to do this and this is covered by this service". I'm not a big fan of things where boilerplate becomes dogma, I'd want to encourage people to think about what they're injecting because they ahve problem X that is solved by Y

Reply:
@peterli-r3 could you please specify what problems injecting these services solves?

Reply:
This flow needs flowEngine because we are utilizing subflows such as finality and CollectSignaturesFlow. 
- needs FlowIdentity because one of the output state needs our own identity. 
- needs FlowMessaging to create flow sessions talking to counterparties. 
- needs TransactionBuilderFactory because this flow builds a transaction. 
- needs IdentityService because this flow involved counterparties. 
- needs NotaryLookupService because the transaction we build in this flow requires notary. 
- need JsonMarshallingService because the in Corda 5 all flows parameters are in the format of JSON.

Reply:
updated. @peterli-r3 could you pls review?